### PR TITLE
cpu/esp*: Xtensa vendor code moved to esp_common

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR contains all the changes required to use the same vendor code for context, exception, and interrupt handling for ESP8266 and ESP32.

### Testing procedure

`tests/thread_basic` should still work for an ESP8266 board as well as an ESP32 board.

### Issues/PRs references

Depends on #10981
This PR is part of refactoring ESP8266 and ESP32 code in issue #10658.

### CODE REVIEW

File: cpu/esp32/thread_arch.c [hunk 1]
```diff
@@ -183,7 +183,7 @@ char* thread_stack_init(thread_task_func_t task_func, void *arg, void *stack_sta
 
     uint32_t *p;
 
-    p = (uint32_t *)(((uint32_t) top_of_stack+1 - XT_CP_SIZE));
+    p = (uint32_t *)(((uint32_t)(top_of_stack + 1) - XT_CP_SIZE) & ~0xf);
```
[the remark this answers is not in the document]
Comment (gschorcht) [reply cpu/esp32/thread_arch.c]: Yes, it was a small difference to the original vendor code which had obviously no affect til now. The address has to be aligned to a multiple of 16. So it is better to have it here.

[the remark this answers is not in the document]
Comment (smlng) [reply cpu/esp32/thread_arch.c]: okay

**Comment by gschorcht** [comment]
@smlng Too late for the 2019.04 release but not too late for next release :smile: I would like to make some progress with this PR (as well as with #10929) since these PRs provide important improvements that are prerequisites for the reimplementation of ESP8266 port in PR #11108.

Sorry that I'm pushing a bit :sunglasses: but the conflicts with current master in PR #11108 are already increasing a lot. It will become more and more difficult to resolve all these conflicts while guaranteeing that the reimplementation PR #11108 keeps as stable as it already was. It was already tested completely.

**Comment by smlng** [comment]
please rebase and squash as needed

**Comment by gschorcht** [comment]
@smIng Thanks for reviewing and testing it again. Rebased and squashed.

**Comment by smlng** [comment]
Side note: careful with my GitHub name, its `smlng` with lower case `L` not `i`, i.e. no vowels. The other guy (`sming`) already got annoyed a couple times 😬

**Comment by smlng** [comment]
CI is not happy, some whitespace stuff, please amend fix directly

**Comment by gschorcht** [comment]
> Side note: careful with my GitHub name, its `smlng` with lower case `L` not `i`, i.e. no vowels. The other guy (`sming`) already got annoyed a couple times

Ups :sunglasses: Usually, I only type in 'sm' and use the completion which is usually 'smlng'. I didn't take much care. I thought Github is intelligent enough to know from the context of the PR that I didn't want to refer 'sming'.

**Comment by gschorcht** [comment]
@smlng Fixed and all light are green.

**Comment by smlng** [comment]
retested, all good!

**Comment by gschorcht** [comment]
@smlng Thanks again for your fast support.

**Comment by smlng** [comment]
> @smlng Thanks again for your fast support.

Sure thing. I now have a couple of ESP lying around at work and home, so I'm happy to help where ever  I can to keep the RIOT support for ESP up and running (and improving)!

**Pull request merged**